src/Host-SDK/Temporal.Worker.Workflows/WorkflowsApi.cs

```diff
@@ -0,0 +1,309 @@
+    }
+
+    // -----------
+
+    public interface IActivityInvocationConfiguration
+    {
+        string TaskQueueMoniker { get; }
+        int ScheduleToStartTimeoutMillisecs { get; }
+        int ScheduleToCloseTimeoutMillisecs { get; }
+        int StartToCloseTimeoutMillisecs { get; }
+        int HeartbeatTimeoutMillisecs { get; }
+        RetryPolicy RetryPolicy { get; }
+    }
+
+    public class ActivityInvocationConfiguration : IActivityInvocationConfiguration
+    {
+        public string TaskQueueMoniker { get; set; }
+        public int ScheduleToStartTimeoutMillisecs { get; set; }
+        public int ScheduleToCloseTimeoutMillisecs { get; set; }
+        public int StartToCloseTimeoutMillisecs { get; set; }
+        public int HeartbeatTimeoutMillisecs { get; set; }
+        public RetryPolicy RetryPolicy { get; set; }
+    }
+
+    // ----------- -----------
+
+    public class RetryPolicy
+    {
+    }
+
+    // ----------- -----------
+
+    /// <summary>
+    /// Specifies that a class is an implementation of a workflow that can be hosted by the worker.
+    /// 
+    /// Can only be applied to classes. Ifaces and structs are not permitted.
+    /// Inheritance is supported.
+    /// If multiple Workflow attributes are present within a class due to inheritance, the most derived wins.
+    /// 
+    /// If 'WorkflowTypeName' is not specified OR null OR Empty OR WhiteSpaceOnly, then the workflow type name is auto-populated
+    /// by taking the class type name.
+    /// 
+    /// 'RunMethod' must be the name of the method that implements them main workflow Run method.
+    /// Such method must have one of the following signatures ("RunAsync" is a placeholder for any method name):
+    ///     Task RunAsync()
+    ///     Task RunAsync(WorkflowContext)
+    ///     Task RunAsync(TArg, WorkflowContext) where TArg : IDataValue
+    ///     Task{TResult} RunAsync() where TResult : IDataValue
+    ///     Task{TResult} RunAsync(WorkflowContext) where TResult : IDataValue
+    ///     Task{TResult} RunAsync(TArg, WorkflowContext) where TResult : IDataValue where TArg : IDataValue
+    ///     Task{PayloadsCollection} RunAsync()
+    ///     Task{PayloadsCollection} RunAsync(WorkflowContext)
+    ///     Task{PayloadsCollection} RunAsync(TArg, WorkflowContext) where TArg : IDataValue    
+    /// otherwise an error during worker initialization is generated.
+    /// 
+    /// If the method named by 'RunMethod' property is overloaded, the property must unambiguously specify a particular overload
+    /// by specifying the full signature. In such a case, either ALL or NONE of the types must use the fully qualified type names.
+    /// E.g., see <c>Part1_5_AttributesAndInterfaces</c>. There it would be sufficient to specify
+    /// 
+    /// <code>
+    ///     [Workflow(runMethod: "ShopAsync")]
```

Comment:
Mentioned before, but I think it's clearer to put an attribute on the method than to be stringly typed here.